src/index.rs

```diff
@@ -95,6 +95,14 @@ impl TreeIndex {
         TreeIndex { height, path: pos }
     }
 
+    /// Construct TreeIndex from a u32 leaf position.
+    pub fn from_u32(height: usize, pos: u32) -> TreeIndex {
+        if height > MAX_HEIGHT {
+            panic!("{}", DecodingError::ExceedMaxHeight);
+        }
+        tree_index_from_u32(height, pos)
+    }
```

Comment:
I wonder if we should have a check here to make sure `pos` is a valid index for a tree of the specified `height`. Something like:
```Rust
if 2.pow(height as u32) <= pos {
    panic!("some message");
}
```

Reply:
Resolved